ClickHouse.Client/Copy/ClickHouseBulkCopy.cs

```diff
@@ -165,13 +169,24 @@ private async Task PushBatch(ICollection<object[]> rows, ClickHouseType[] column
                 }
 
                 using var writer = new ExtendedBinaryWriter(gzipStream);
-                foreach (var row in rows)
+                object[] excp_rows = null;
```

Comment:
You don't need a variable to hold the last row, you have "last" row index in the collection